DCOS/autoscale-testing/AutoScale.Tests.ps1

```diff
@@ -67,6 +67,39 @@ function AreAllNodesHealthy ($RG_NAME) {
     return $isHealthy
 }
 
+function AllMetricsGood ($RG_NAME) {
+    $masterFQDN = getMasterFQDN($RG_NAME)
+    $res = Invoke-WebRequest "http://$masterFQDN/dcos-history-service/history/last" | ConvertFrom-Json | Select-Object slaves
+    $agentCount = $res.slaves.Count
+    $isAllGood = $true
+    foreach ($agentNode in $res.slaves) {
+        Try
+        {
+            $agentMetricsRequestEp = "http://$masterFQDN/system/v1/agent/$($agentNode.id)/metrics/v0/node"
+            $dataPoints = Invoke-WebRequest $agentMetricsRequestEp | ConvertFrom-Json 
+            $dataPointCount = $dataPoints.datapoints.Count
+
+            if ($dataPointCount -eq 0) {
+                Write-Output "Got no datapoints back, somthing wrong with dcos-metrics service"
+                Write-Output "agentNode.attributes.os = $($agentNode.attributes.os)"
+                Write-Output "agentNode.id = $($agentNode.id)"
+                Write-Output "agentNode.hostname = $($agentNode.hostname)"
+                Write-Output "dataPointCount = $dataPointCount"
+                $isAllGood = $false
+            }
+        }
+        Catch
+        {
+            $ErrorMessage = $_.Exception.Message
+            Write-Output "$ErrorMessage, metrics query failed on agent: hostname=$($agentNode.hostname) osType = $osType id= $($agentNode.id) "
```

Comment:
` $osType` is not defined over here. I think you wanted to use: `$($agentNode.attributes.os)`

Reply:
I should have done that when removing $osType variable from the code during cleanup.
I will fix this.